exampleoc/structs-0.go

```diff
@@ -1535,6 +1536,104 @@ type Model_SingleKey struct {
 // identify it as being generated by ygen.
 func (*Model_SingleKey) IsYANGGoStruct() {}
 
+// NewSingleKey creates a new entry in the SingleKey list of the
+// Model_SingleKey struct. The keys of the list are populated from the input
+// arguments.
+func (t *Model_SingleKey) NewSingleKey(Key string) (*Model_SingleKey_SingleKey, error) {
+
+	// Initialise the list within the receiver struct if it has not already been
+	// created.
+	if t.SingleKey == nil {
+		t.SingleKey = make(map[string]*Model_SingleKey_SingleKey)
+	}
+
+	key := Key
+
+	// Ensure that this key has not already been used in the
+	// list. Keyed YANG lists do not allow duplicate keys to
+	// be created.
+	if _, ok := t.SingleKey[key]; ok {
+		return nil, fmt.Errorf("duplicate key %v for list SingleKey", key)
+	}
+
+	t.SingleKey[key] = &Model_SingleKey_SingleKey{
+		Key: &Key,
+	}
+
+	return t.SingleKey[key], nil
```

Comment:
(no action req): where's all this whitespace coming from? it'd be nice to reduce it

Reply:
ygot, feel free to change the templates there.